compiler/codegen/symbol.cpp

```diff
@@ -1716,6 +1740,73 @@ GenRet FnSymbol::codegenPXDType() {
   return ret;
 }
 
+// Supports the creation of a python module with --library-python
+GenRet FnSymbol::codegenPYXType() {
+  GenInfo* info = gGenInfo;
+  GenRet ret;
+
+  ret.chplType = typeInfo();
+
+  if (info->cfile) {
+    // Function header
+    std::string header = "def ";
+    header += cname;
+    header += "(";
+
+    // Translation of any arguments with Python-specific types
+    std::string argTranslate = "";
+    // Call to the wrapped function
+    std::string funcCall = "\t";
+    // Return statement, if applicable
+    std::string returnStmt = "";
+    if (retType != dtVoid) {
+      funcCall += "ret = ";
+      returnStmt += "\treturn ret\n";
+    }
+    funcCall += "chpl_";
+    funcCall += cname;
+    funcCall += "(";
+
+    if (numFormals() != 0) {
+      int count = 0;
+      for_formals(formal, this) {
+        if (formal->hasFlag(FLAG_NO_CODEGEN))
+          continue; // do not print locale argument, end count, dummy class
+        if (count > 0) {
+          header += ", ";
+          funcCall += ", ";
+        }
+
+        header += formal->getPythonType(false);
+        header += " ";
+        header += formal->cname;
+        if (fGenIDS) {
+          header += " ";
+          header += idCommentTemp(formal);
+        }
+
+        std::string curArgTranslate = formal->getPythonArgTranslation();
+        if (curArgTranslate != "") {
+          argTranslate += curArgTranslate;
+          funcCall += "chpl_";
+        }
+
+        funcCall += formal->cname;
+        count++;
+      }
+
+    } // pyx files do not take void as an argument list, just close the parens
+    header += "):\n";
+    funcCall += ")\n";
+    ret.c = header + argTranslate + funcCall + returnStmt;
+
+  } else {
+    // TODO: LLVM stuff
```

Comment:
So these LLVM TODOs are getting a bit ridiculous. Supposing, for the sake of argument, that we were using --llvm to build a library and wanted the compiler to generate these Cython interop files. Then info->cfile would be set to the Cython file, right? Same thing with the header file generation. Even if we use --llvm, we still want to emit C declarations for the header, so info->cfile should be set.

Reply:
That makes sense.  I remember thinking something along those lines at some point so I'm surprised this one survived, I'll remove it.